core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -1141,7 +1169,7 @@ class DAGSchedulerSuite extends SparkFunSuite with LocalSparkContext with TimeLi
       taskSets(0).tasks(1),
       TaskKilled("test"),
       null))
-    assert(failedStages === Seq(0))
+    assert(sparkListener.failedStages === Seq(0))
```

Comment:
This is actual place of bug. The code is newly added (for "newly" I meant after the pattern of `sc.listenerBus.waitUntilEmpty(WAIT_TIMEOUT_MILLIS)` is applied) and doesn't follow the existing pattern. That is easily missed and here we can enforce it by disallowing access variables directly.

Reply:
There're some other places accessing failedStages without waiting. Search with `assert(failedStages === Seq(0))`